add github issue templates (#1859)
prisma/prisma: .github/ISSUE_TEMPLATE/bug_report.md

---
name: Bug report
about: Create a bug report to help us improve Prisma
title: ''
labels: ''
assignees: ''

---

<!-- 
Thanks for helping us improve Prisma! 🙏 Please follow the sections in the template and provide as much information as possible about your problem, e.g. by setting the `DEBUG="*"` env var and enabling additional logging output in Prisma Client.

Learn more about writing proper bug reports here: https://pris.ly/d/bug-reports
--> 

> ## Describe the bug

Reviewer: Consistent wording:
```suggestion
## Bug Description
```

<!-- A clear and concise description of what the bug is. -->

## How to reproduce
<!-- 
Steps to reproduce the behavior:
1. Go to '...'
2. Change '....'
3. Run '....'
4. See error 
-->

## Expected behavior
<!-- A clear and concise description of what you expected to happen. -->

> ## Additional Prisma info (Prisma schema, Prisma Client queries, ...)

Reviewer: ```suggestion
### Prisma information 
<!-- Prisma schema, Prisma Client code and queries, ... -->
```

<!-- Do not include your database credentials when sharing your Prisma schema! -->

## Environment & setup
<!-- In which environment does the problem occur -->

- OS: [e.g. Mac OS, Windows, Debian, CentOS, ...]

> - Database: [PostgreSQL, MySQL or SQLite]

Reviewer 2: We may want to add MariaDB here

Author: Done

---
name: Bug report

> about: Create a report to help us improve

Reviewer 3: ```suggestion
about: Create a bug report to help us improve Prisma
```

title: ''
labels: ''
assignees: ''

---

<!-- 

> Thanks for helping us improve Prisma! 🙏 Please follow the sections in the template and provide as much information as possible about your problem, e.g. by setting the `DEBUG="*"` env var and enabling additional logging output in Prisma Client.

Reviewer: nitpick/suggestion
```suggestion
Thanks for helping us improve Prisma! 🙏 Please follow the sections in the template and provide as much information as possible about your problem, e.g. by setting the `DEBUG="*"` environment variable and enabling additional logging output in Prisma Client.
```

Learn more about writing proper bug reports here: https://pris.ly/d/bug-reports
--> 

## Describe the bug
<!-- A clear and concise description of what the bug is. -->

## How to reproduce
<!-- 
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'

> 4. See error 

Reviewer 3: Nit: Although fine, but these steps look like they were meant for a web app.

```suggestion
1. Go to '...'
2. Change '....'
3. Run '....'
4. See error 
```

-->

## Expected behavior
<!-- A clear and concise description of what you expected to happen. -->

## Additional Prisma info (Prisma schema, Prisma Client queries, ...)
<!-- Do not include your database credentials when sharing your Prisma schema! -->

## Environment & setup
<!-- In which environment does the problem occur -->

- OS: [e.g. Mac OS, Windows, Debian, CentOS, ...]
- Database: [PostgreSQL, MySQL or SQLite]
- Prisma version: [Run `prisma2 -v` to see your Prisma version]

> - Node.js version: [Run `node -v` to see your Node.js version]

Reviewer: ```suggestion
## Bug Description
<!-- A clear and concise description of what the bug is. -->

### How to reproduce
<!-- 
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error 
-->

### Expected behavior
<!-- A clear and concise description of what you expected to happen. -->

## Information

### Prisma information 
<!-- Prisma schema, Prisma Client code and queries, ... -->
<!-- Do not include your database credentials when sharing your Prisma schema! -->

### Environment & setup
<!-- In which environment does the problem occur -->

- OS: [e.g. Mac OS, Windows, Debian, CentOS, ...]
- Database: [PostgreSQL, MySQL or SQLite]
- Prisma version: [Run `prisma2 -v` to see your Prisma version]
- Node.js version: [Run `node -v` to see your Node.js version]
```

Author: Not a _huge_ fan of this change. I'm not sure this is adding a lot of value but rather makes the template slightly more complicated. Also, right now the fact that "Bug description" doesn't have its own `##` heading could be misinterpreted that people should only provide information for the `###` sections. I think I prefer the previous version but happy to discuss further.

Reviewer: > Also, right now the fact that "Bug description" doesn't have its own `##` heading could be misinterpreted that people should only provide information for the `###` sections.

This was a nod for you asking for not adding more headlines - I would be fine having an additional `### Information` or something like that if the hierarchy is that important.

IMO structure trumps the need for pretty much everything else, as it leads to clear thinking (or at least encourages it).

Author: > IMO structure trumps the need for pretty much everything else, as it leads to clear thinking (or at least encourages it).

Generally agree but I'm wary of adding "sub-sections" (mostly because I don't expect everyone creating issues to know Markdown and these additional sections might cause confusion in that case). I'm still not convinced these structural changes add more value, IMO the template and what we're asking is pretty clear as it stands.
